### PR TITLE
First version

### PR DESCRIPTION
**Pull request by dburrowes**
PROBLEM
===
We could/should move our logging to a consistent logging module (with an eye to eventually connecting that to angular, delivering the information to the server, etc).

THIS PROBLEM
====
This is the general purpose dxLog module.  It will be consumed by the tranformation of dxData that is going on.

DISCUSSION
===
After the previous review, I went back and decided to revisit some of my assumptions, and also move the piece, so to speak, another couple steps ahead of where I was thinking at the time.
Thus, this provides a full dxLog module, with all the build etc processes to support that.  

Some things to note about this:
* Do note that this is a source tree (a git repo), whose source directory contains information to build (among other things) an npm module.  Thus, you'll find a package.json, README, etc. for both the source tree and as a source file that will be added to the distribution npm module.
* The build process will run our style checks, unit tests and code coverage against this code, report all this to the client, and then assemble the code both as an npm module and as an es5 javascript file

-----

~/dburrowes/dxLog [master]: grunt
Running "eslint:target" (eslint) task

Running "jscs:src" (jscs) task
>> 7 files without code style errors.

Running "karma:dist" (karma) task
Safari 9.0.2 (Mac OS X 10.11.2): Executed 10 of 10 SUCCESS (0.039 secs / 0.004 secs)
PhantomJS 2.0.0 (Mac OS X 0.0.0): Executed 10 of 10 SUCCESS (0.008 secs / 0.013 secs)
Chrome 47.0.2526 (Mac OS X 10.11.2): Executed 10 of 10 SUCCESS (0.031 secs / 0.009 secs)
Firefox 42.0.0 (Mac OS X 10.11.0): Executed 10 of 10 SUCCESS (0.009 secs / 0.006 secs)
TOTAL: 40 SUCCESS

Running "_testreport" task
Style Checks passed
Coverge summary:

Running "cat:testReport" (cat) task
-----------|----------|----------|----------|----------|----------------|
File       |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-----------|----------|----------|----------|----------|----------------|
 source/   |      100 |      100 |      100 |      100 |                |
  index.js |      100 |      100 |      100 |      100 |                |
-----------|----------|----------|----------|----------|----------------|
All files  |      100 |      100 |      100 |      100 |                |
-----------|----------|----------|----------|----------|----------------|

IE 9.0.0 (Windows 7 0.0.0): Executed 10 of 10 SUCCESS (0.133 secs / 0.005 secs)
Chrome 47.0.2526 (Windows 7 0.0.0): Executed 10 of 10 SUCCESS (0.123 secs / 0.012 secs)
Firefox 43.0.0 (Windows 7 0.0.0): Executed 10 of 10 SUCCESS (0.237 secs / 0.008secs)
PhantomJS 2.0.0 (Windows 7 0.0.0): Executed 10 of 10 SUCCESS (0.078 secs / 0.004 secs)
TOTAL: 30 SUCCESS

Running "_testreport" task
Style Checks passed
Coverge summary:

Running "cat:testReport" (cat) task
-----------|----------|----------|----------|----------|----------------|
File       |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-----------|----------|----------|----------|----------|----------------|
 source\   |      100 |      100 |      100 |      100 |                |
  index.js |      100 |      100 |      100 |      100 |                |
-----------|----------|----------|----------|----------|----------------|
All files  |      100 |      100 |      100 |      100 |                |
-----------|----------|----------|----------|----------|----------------|

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright (c) 2015, 2016 by Delphix. All rights reserved.
```
Comment (blazarus) [inline Gruntfile.js]: Is it common for open source projects to have a copyright like this? What happens if someone outside of Delphix contributes?

File: Gruntfile.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+var DIST_DIR = './dist/';
+var NODE_MODULES_DIR = './node_modules';
+var DIST_NPM_DIR = DIST_DIR + '/npm/dxLog/';
+var DIST_FLAT_DIR = DIST_DIR + '/es5file/';
+
+var JS_FILE_GLOB = [ SOURCE_DIR + '/*.js', '!' + SOURCE_DIR + '/*.unit.js'];
+var JS_AND_TEST_FILE_GLOB = [ SOURCE_DIR + '/*.js'];
+var OTHER_FILE_GLOB = [ SOURCE_DIR + '/*', '!' + SOURCE_DIR + '/*.js'];
+
+
```
Comment (blazarus) [inline Gruntfile.js]: Extra new line. Are you running the style checks on these files?

p.s. wow can you not select multiple lines for a comment in github reviews? That's annoying.

File: Gruntfile.js (hunk 1 above)
Comment (bodawei) [inline Gruntfile.js]: Interestingly, it is. See for comparison: https://github.com/jashkenas/underscore/blob/master/underscore-min.js

Comment (blazarus) [inline Gruntfile.js]: Huh. Fair enough.

File: Gruntfile.js (hunk 2 above)
Comment (bodawei) [inline Gruntfile.js]: I don't think I had grabbed the wrong diff. But I've been fiddling with the build process some, so may have regressed that.  I've got a new fix.  Now need to determine how to update this pull request.  Definitely not as smooth as review board.

If you like the change can you say "Pull It" so I have what amounts to a formal second pair of eyes saying it is good?

Comment (bodawei) [inline Gruntfile.js]: Ah, I see what you meant.  I'd grabbed the right version, from my point of view. but the internal was, conceptually, on a branch which I'd pruned with the decision to move review here. I've now moved those changes here too.

**Comment by blazarus** [comment]
pull it!

**Pull request merged**